trustmanager/keystore_test.go

```diff
@@ -89,16 +88,18 @@ func TestKeyStoreInternalState(t *testing.T) {
 		privKey, err := utils.GenerateECDSAKey(rand.Reader)
 		require.NoError(t, err, "could not generate private key")
 
+		var privKeyPEM []byte
 		// generate the correct PEM role header
-		privKeyPEM, err := utils.KeyToPEM(privKey, role)
+		if role == data.CanonicalRootRole || data.IsDelegation(role) || !data.ValidRole(role) {
+			privKeyPEM, err = utils.KeyToPEM(privKey, role, "")
+		} else {
+			privKeyPEM, err = utils.KeyToPEM(privKey, role, gun)
+		}
+
 		require.NoError(t, err, "could not generate PEM")
 
 		// write the key file to the correct location
-		// Pretend our GUN is docker.com/notary
-		keyPath := filepath.Join(tempBaseDir, "private", getSubdir(role))
-		if role == data.CanonicalTargetsRole || role == data.CanonicalSnapshotRole {
-			keyPath = filepath.Join(keyPath, gun)
-		}
+		keyPath := filepath.Join(tempBaseDir, "private")
```

Comment:
We can also use `notary.RootKeysSubdir` and `notary.NonRootKeysSubdir` instead of `"root_keys"` and `"tuf_keys"`, respectively